src/test/java/uk/gov/companieshouse/chipsrestinterfacesconsumer/service/impl/MessageProcessorServiceImplTest.java

```diff
@@ -19,18 +24,29 @@
     @Mock
     private ChipsRestClient chipsRestClient;
 
+    @Mock
+    private MessageProducer retryMessageProducer;
+
     @Mock
     private ApplicationLogger logger;
 
     @InjectMocks
     private MessageProcessorServiceImpl messageProcessorService;
 
     @Test
-    void processMessageTest() {
+    void processMessageTest() throws ServiceException {
         ChipsKafkaMessage chipsKafkaMessage = new ChipsKafkaMessage();
 
         messageProcessorService.processMessage(chipsKafkaMessage);
 
         verify(chipsRestClient, times(1)).sendToChips(eq(chipsKafkaMessage));
     }
+
+    @Test
+    void testRetryIsCalled() throws ServiceException {
+        ChipsKafkaMessage chipsKafkaMessage = new ChipsKafkaMessage();
+        doThrow(RestClientException.class).when(chipsRestClient).sendToChips(chipsKafkaMessage);
+        messageProcessorService.processMessage(chipsKafkaMessage);
+        verify(retryMessageProducer, times(1)).writeToQueue(eq(chipsKafkaMessage));
```

Comment:
as it should be either/or for which route to take (chips or retry), I would also include a verify times(0) / never for the chipsClient to verify it is not called.

Reply:
That will be called.

Reply:
ok, you should verify it is called as well

Reply:
ah, sorry i can see it is throwing an exception above, forget that.

Reply:
ok